syntax = "proto3";

import "google/protobuf/timestamp.proto";

// The greeting service definition.
service Greeter {
  // Sends a greeting
  rpc SayHello (HelloRequest) returns (HelloReply) {}
}

// The request message containing the user's name.
message HelloRequest {
  string name = 1;
  repeated ShelfGrow shelf_grows = 2;
  GrowPhase grow_phase = 3;
  int32 power_level = 4;
  int32 red_level = 5;
  int32 blue_level = 6;
}

// The response message containing the greetings
message HelloReply {
  string message = 1;
}

message ShelfGrow {
  int32 grow_id = 1;
  int32 room_id = 2;
  int32 rack_id = 3;
  int32 shelf_id = 4;
}

message GrowPhase {
  int32 grow_id = 1;
  int32 recipe_phase_num = 2;
  int32 recipe_id = 3;
  google.protobuf.Timestamp phase_start_datetime = 4;
  google.protobuf.Timestamp phase_end_datetime = 5;
  bool is_last_phase = 6;
}